### PR TITLE
DAOS-4542 dtx: (1) remove dae_layout_gen from DTX entry

### PR DESCRIPTION
**Pull request by Nasf-Fan**
It was used for deregister DTX entry when container to be destroyed.
Under such case, the DTX entry in DRAM has been removed, the on-disk
DTX entry will be destroyed soon via vos_dtx_table_destroy(). So do
not need to care about the DTX entry layout generation changes.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

File: src/vos/vos_layout.h [hunk 1]
```diff
@@ -179,8 +179,6 @@ struct vos_dtx_act_ent_df {
 	/** The server generation when handles the DTX. */
 	uint64_t			dae_srv_gen;
 	/** The active DTX entry on-disk layout generation. */
```
Comment (jolivier23) [inline src/vos/vos_layout.h:181]: You removed the wrong comment

**Comment by Nasf-Fan** [comment]
> https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-2691/7/testReport/junit/(root)/IorSmall/Test___Functional_Hardware_Large___1___io_ior_small_py_IorSmall_test_ior_small_container_dfuse_hosts_np_16_iorflags_pool_servers_server_manager_fd22/

> https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-2691/7/testReport/junit/(root)/NvmeObject/Test___Functional_Hardware_Large___1___io_nvme_object_py_NvmeObject_test_nvme_object_single_pool_container_hosts_createmode_createset_createsize_createsvc_servers_server_manager_timeouts_4cea/

> https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-2691/7/testReport/junit/(root)/MdtestSmall/Test___Functional_Hardware_Large___1___io_mdtest_small_py_MdtestSmall_test_mdtest_small_container_dfuse_hosts_np_8_pool_server_config_d209/


ior_small, nvme_object and mdtest_small failed because of DAOS-4665.

**Comment by Nasf-Fan** [comment]
> https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2691/7/artifact/Functional/job-2020-05-19T03.50-4fb1bea-io-hdf5/job.log/*view*/

hdf5 timeout because of DAOS-4471.

**Comment by Nasf-Fan** [comment]
CI test failures are caused by known issues, suggest to force land it.

**Pull request merged**